### PR TITLE
Fix multiple spellings in TaskPool documentation

### PR DESCRIPTION
**Pull request by aggarg**
Description
-----------
This commit also fixes one minor initialization issue where memset was
not passed the correct length. Other than than all the changes are in
comments only.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

File: modules/libraries/standard/common/include/iot_taskpool.h [hunk 1]
```diff
@@ -82,7 +82,7 @@
  *
  * This function should be called once by the application to initialize the one single instance of the system task pool.
  * An application should initialize the system task pool early in the boot sequence, before initializing any other library
- * that uses the system task pool. such as MQTT, Shadow, Defernder, etc.. An application should also initialize the system
+ * that uses the system task pool. such as MQTT, Shadow, Defender, etc.. An application should also initialize the system
```
Comment (dcgaws) [inline modules/libraries/standard/common/include/iot_taskpool.h]: There are still typos in this line. Can we just say "... sequence, before initializing any other library. An application should also..."? Giving a partial list of example libraries is probably misleading.

File: modules/libraries/standard/common/taskpool/iot_taskpool.c [hunk 2]
```diff
@@ -1399,7 +1401,7 @@ static void _signalShutdown( _taskPool_t * const pTaskPool,
     /* Set the exit condition. */
     pTaskPool->maxThreads = 0;
 
-    /* Broadcast to all active threads to wake-up. Active threads do check the exit condition right after wakein up. */
+    /* Broadcast to all active threads to wake-up. Active threads do check the exit condition right after wakeing up. */
```
Comment (dcgaws) [inline modules/libraries/standard/common/taskpool/iot_taskpool.c]: waking

Comment (aggarg) [reply modules/libraries/standard/common/taskpool/iot_taskpool.c]: Done.

File: modules/libraries/standard/common/include/iot_taskpool.h (hunk 1 above)
Comment (aggarg) [reply modules/libraries/standard/common/include/iot_taskpool.h]: Fixed.

**Pull request closed without merging**